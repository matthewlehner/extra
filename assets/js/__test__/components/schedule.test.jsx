import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Schedule from "components/schedule";

describe("Schedule component", () => {
  it("renders correctly", () => {
    const props = {
      toggleAutopilot: jest.fn(),
      schedule: {
        autopilot: true,
        timeslots: []
      }
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Schedule {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
