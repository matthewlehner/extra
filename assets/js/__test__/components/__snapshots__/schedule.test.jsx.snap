// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule component renders correctly 1`] = `
<section
  className="channel-schedule"
>
  <form
    className="form underlinedForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="schedule[timezone]"
    >
      Timezone
    </label>
    <div
      className="select"
    >
      <select
        className="form__control"
        id="schedule[timezone]"
        name="timezone"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
           
        </option>
        <option
          value="Pacific/Midway"
        >
          (UTC-11:00) Midway Island, American Samoa
        </option>
        <option
          value="America/Adak"
        >
          (UTC-10:00) Aleutian Islands
        </option>
        <option
          value="Pacific/Honolulu"
        >
          (UTC-10:00) Hawaii
        </option>
        <option
          value="Pacific/Marquesas"
        >
          (UTC-09:30) Marquesas Islands
        </option>
        <option
          value="America/Anchorage"
        >
          (UTC-09:00) Alaska
        </option>
        <option
          value="America/Tijuana"
        >
          (UTC-08:00) Baja California
        </option>
        <option
          value="America/Vancouver"
        >
          (UTC-08:00) Pacific Time (US and Canada)
        </option>
        <option
          value="America/Phoenix"
        >
          (UTC-07:00) Arizona
        </option>
        <option
          value="America/Chihuahua"
        >
          (UTC-07:00) Chihuahua, La Paz, Mazatlan
        </option>
        <option
          value="America/Denver"
        >
          (UTC-07:00) Mountain Time (US and Canada)
        </option>
        <option
          value="America/Belize"
        >
          (UTC-06:00) Central America
        </option>
        <option
          value="America/Chicago"
        >
          (UTC-06:00) Central Time (US and Canada)
        </option>
        <option
          value="Pacific/Easter"
        >
          (UTC-06:00) Easter Island
        </option>
        <option
          value="America/Mexico_City"
        >
          (UTC-06:00) Guadalajara, Mexico City, Monterrey
        </option>
        <option
          value="America/Regina"
        >
          (UTC-06:00) Saskatchewan
        </option>
        <option
          value="America/Bogota"
        >
          (UTC-05:00) Bogota, Lima, Quito
        </option>
        <option
          value="America/Cancun"
        >
          (UTC-05:00) Chetumal
        </option>
        <option
          value="America/New_York"
        >
          (UTC-05:00) Eastern Time (US and Canada)
        </option>
        <option
          value="America/Port-au-Prince"
        >
          (UTC-05:00) Haiti
        </option>
        <option
          value="America/Havana"
        >
          (UTC-05:00) Havana
        </option>
        <option
          value="America/Indiana/Indianapolis"
        >
          (UTC-05:00) Indiana (East)
        </option>
        <option
          value="America/Asuncion"
        >
          (UTC-04:00) Asuncion
        </option>
        <option
          value="America/Halifax"
        >
          (UTC-04:00) Atlantic Time (Canada)
        </option>
        <option
          value="America/Caracas"
        >
          (UTC-04:00) Caracas
        </option>
        <option
          value="America/Cuiaba"
        >
          (UTC-04:00) Cuiaba
        </option>
        <option
          value="America/Manaus"
        >
          (UTC-04:00) Georgetown, La Paz, Manaus, San Juan
        </option>
        <option
          value="America/Santiago"
        >
          (UTC-04:00) Santiago
        </option>
        <option
          value="America/Grand_Turk"
        >
          (UTC-04:00) Turks and Caicos
        </option>
        <option
          value="America/St_Johns"
        >
          (UTC-03:30) Newfoundland
        </option>
        <option
          value="America/Fortaleza"
        >
          (UTC-03:00) Araguaina
        </option>
        <option
          value="America/Sao_Paulo"
        >
          (UTC-03:00) Brasilia
        </option>
        <option
          value="America/Cayenne"
        >
          (UTC-03:00) Cayenne, Fortaleza
        </option>
        <option
          value="America/Buenos_Aires"
        >
          (UTC-03:00) City of Buenos Aires
        </option>
        <option
          value="America/Godthab"
        >
          (UTC-03:00) Greenland
        </option>
        <option
          value="America/Montevideo"
        >
          (UTC-03:00) Montevideo
        </option>
        <option
          value="America/Miquelon"
        >
          (UTC-03:00) Saint Pierre and Miquelon
        </option>
        <option
          value="America/Bahia"
        >
          (UTC-03:00) Salvador
        </option>
        <option
          value="America/Noronha"
        >
          (UTC-02:00) Fernando de Noronha
        </option>
        <option
          value="Atlantic/Azores"
        >
          (UTC-01:00) Azores
        </option>
        <option
          value="Atlantic/Cape_Verde"
        >
          (UTC-01:00) Cabo Verde Islands
        </option>
        <option
          value="Africa/Casablanca"
        >
          (UTC) Casablanca
        </option>
        <option
          value="Europe/London"
        >
          (UTC) Dublin, Edinburgh, Lisbon, London
        </option>
        <option
          value="Africa/Monrovia"
        >
          (UTC) Monrovia, Reykjavik
        </option>
        <option
          value="Europe/Amsterdam"
        >
          (UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
        </option>
        <option
          value="Europe/Belgrade"
        >
          (UTC+01:00) Belgrade, Bratislava, Budapest, Ljubljana, Prague
        </option>
        <option
          value="Europe/Brussels"
        >
          (UTC+01:00) Brussels, Copenhagen, Madrid, Paris
        </option>
        <option
          value="Europe/Warsaw"
        >
          (UTC+01:00) Sarajevo, Skopje, Warsaw, Zagreb
        </option>
        <option
          value="Africa/Algiers"
        >
          (UTC+01:00) West Central Africa
        </option>
        <option
          value="Africa/Windhoek"
        >
          (UTC+01:00) Windhoek
        </option>
        <option
          value="Asia/Amman"
        >
          (UTC+02:00) Amman
        </option>
        <option
          value="Europe/Athens"
        >
          (UTC+02:00) Athens, Bucharest
        </option>
        <option
          value="Asia/Beirut"
        >
          (UTC+02:00) Beirut
        </option>
        <option
          value="Africa/Cairo"
        >
          (UTC+02:00) Cairo
        </option>
        <option
          value="Asia/Damascus"
        >
          (UTC+02:00) Damascus
        </option>
        <option
          value="Asia/Gaza"
        >
          (UTC+02:00) Gaza, Hebron
        </option>
        <option
          value="Africa/Harare"
        >
          (UTC+02:00) Harare, Pretoria
        </option>
        <option
          value="Europe/Helsinki"
        >
          (UTC+02:00) Helsinki, Kiev, Riga, Sofia, Tallinn, Vilnius
        </option>
        <option
          value="Asia/Jerusalem"
        >
          (UTC+02:00) Jerusalem
        </option>
        <option
          value="Europe/Kaliningrad"
        >
          (UTC+02:00) Kaliningrad
        </option>
        <option
          value="Africa/Tripoli"
        >
          (UTC+02:00) Tripoli
        </option>
        <option
          value="Asia/Baghdad"
        >
          (UTC+03:00) Baghdad
        </option>
        <option
          value="Asia/Istanbul"
        >
          (UTC+03:00) Istanbul
        </option>
        <option
          value="Asia/Kuwait"
        >
          (UTC+03:00) Kuwait, Riyadh
        </option>
        <option
          value="Europe/Minsk"
        >
          (UTC+03:00) Minsk
        </option>
        <option
          value="Europe/Moscow"
        >
          (UTC+03:00) Moscow, St. Petersburg, Volgograd
        </option>
        <option
          value="Africa/Nairobi"
        >
          (UTC+03:00) Nairobi
        </option>
        <option
          value="Asia/Tehran"
        >
          (UTC+03:30) Tehran
        </option>
        <option
          value="Asia/Muscat"
        >
          (UTC+04:00) Abu Dhabi, Muscat
        </option>
        <option
          value="Europe/Astrakhan"
        >
          (UTC+04:00) Astrakhan, Ulyanovsk
        </option>
        <option
          value="Asia/Baku"
        >
          (UTC+04:00) Baku
        </option>
        <option
          value="Europe/Samara"
        >
          (UTC+04:00) Izhevsk, Samara
        </option>
        <option
          value="Indian/Mauritius"
        >
          (UTC+04:00) Port Louis
        </option>
        <option
          value="Asia/Tbilisi"
        >
          (UTC+04:00) Tbilisi
        </option>
        <option
          value="Asia/Yerevan"
        >
          (UTC+04:00) Yerevan
        </option>
        <option
          value="Asia/Kabul"
        >
          (UTC+04:30) Kabul
        </option>
        <option
          value="Asia/Tashkent"
        >
          (UTC+05:00) Tashkent, Ashgabat
        </option>
        <option
          value="Asia/Yekaterinburg"
        >
          (UTC+05:00) Ekaterinburg
        </option>
        <option
          value="Asia/Karachi"
        >
          (UTC+05:00) Islamabad, Karachi
        </option>
        <option
          value="Asia/Kolkata"
        >
          (UTC+05:30) Chennai, Kolkata, Mumbai, New Delhi
        </option>
        <option
          value="Asia/Colombo"
        >
          (UTC+05:30) Sri Jayawardenepura
        </option>
        <option
          value="Asia/Katmandu"
        >
          (UTC+05:45) Kathmandu
        </option>
        <option
          value="Asia/Almaty"
        >
          (UTC+06:00) Astana
        </option>
        <option
          value="Asia/Dhaka"
        >
          (UTC+06:00) Dhaka
        </option>
        <option
          value="Asia/Rangoon"
        >
          (UTC+06:30) Yangon (Rangoon)
        </option>
        <option
          value="Asia/Novosibirsk"
        >
          (UTC+07:00) Novosibirsk
        </option>
        <option
          value="Asia/Bangkok"
        >
          (UTC+07:00) Bangkok, Hanoi, Jakarta
        </option>
        <option
          value="Asia/Barnaul"
        >
          (UTC+07:00) Barnaul, Gorno-Altaysk
        </option>
        <option
          value="Asia/Hovd"
        >
          (UTC+07:00) Hovd
        </option>
        <option
          value="Asia/Krasnoyarsk"
        >
          (UTC+07:00) Krasnoyarsk
        </option>
        <option
          value="Asia/Tomsk"
        >
          (UTC+07:00) Tomsk
        </option>
        <option
          value="Asia/Chongqing"
        >
          (UTC+08:00) Beijing, Chongqing, Hong Kong SAR, Urumqi
        </option>
        <option
          value="Asia/Irkutsk"
        >
          (UTC+08:00) Irkutsk
        </option>
        <option
          value="Asia/Kuala_Lumpur"
        >
          (UTC+08:00) Kuala Lumpur, Singapore
        </option>
        <option
          value="Australia/Perth"
        >
          (UTC+08:00) Perth
        </option>
        <option
          value="Asia/Taipei"
        >
          (UTC+08:00) Taipei
        </option>
        <option
          value="Asia/Ulaanbaatar"
        >
          (UTC+08:00) Ulaanbaatar
        </option>
        <option
          value="Asia/Pyongyang"
        >
          (UTC+08:30) Pyongyang
        </option>
        <option
          value="Australia/Eucla"
        >
          (UTC+08:45) Eucla
        </option>
        <option
          value="Asia/Chita"
        >
          (UTC+09:00) Chita
        </option>
        <option
          value="Asia/Tokyo"
        >
          (UTC+09:00) Osaka, Sapporo, Tokyo
        </option>
        <option
          value="Asia/Seoul"
        >
          (UTC+09:00) Seoul
        </option>
        <option
          value="Asia/Yakutsk"
        >
          (UTC+09:00) Yakutsk
        </option>
        <option
          value="Australia/Adelaide"
        >
          (UTC+09:30) Adelaide
        </option>
        <option
          value="Australia/Darwin"
        >
          (UTC+09:30) Darwin
        </option>
        <option
          value="Australia/Brisbane"
        >
          (UTC+10:00) Brisbane
        </option>
        <option
          value="Australia/Canberra"
        >
          (UTC+10:00) Canberra, Melbourne, Sydney
        </option>
        <option
          value="Pacific/Guam"
        >
          (UTC+10:00) Guam, Port Moresby
        </option>
        <option
          value="Australia/Hobart"
        >
          (UTC+10:00) Hobart
        </option>
        <option
          value="Asia/Vladivostok"
        >
          (UTC+10:00) Vladivostok
        </option>
        <option
          value="Australia/Lord_Howe"
        >
          (UTC+10:30) Lord Howe Island
        </option>
        <option
          value="Pacific/Bougainville"
        >
          (UTC+11:00) Bougainville Island
        </option>
        <option
          value="Asia/Srednekolymsk"
        >
          (UTC+11:00) Chokirdakh
        </option>
        <option
          value="Asia/Magadan"
        >
          (UTC+11:00) Magadan
        </option>
        <option
          value="Pacific/Norfolk"
        >
          (UTC+11:00) Norfolk Island
        </option>
        <option
          value="Asia/Sakhalin"
        >
          (UTC+11:00) Sakhalin
        </option>
        <option
          value="Pacific/Guadalcanal"
        >
          (UTC+11:00) Solomon Islands, New Caledonia
        </option>
        <option
          value="Asia/Anadyr"
        >
          (UTC+12:00) Anadyr, Petropavlovsk-Kamchatsky
        </option>
        <option
          value="Pacific/Auckland"
        >
          (UTC+12:00) Auckland, Wellington
        </option>
        <option
          value="Pacific/Fiji"
        >
          (UTC+12:00) Fiji Islands
        </option>
        <option
          value="Pacific/Chatham"
        >
          (UTC+12:45) Chatham Islands
        </option>
        <option
          value="Pacific/Tongatapu"
        >
          (UTC+13:00) Nuku'alofa
        </option>
        <option
          value="Pacific/Apia"
        >
          (UTC+13:00) Samoa
        </option>
        <option
          value="Pacific/Kiritimati"
        >
          (UTC+14:00) Kiritimati Island
        </option>
      </select>
    </div>
  </form>
  <div
    className="tab-container"
  >
    <nav
      className="tablist"
      role="tablist"
    >
      <a
        aria-controls="monday"
        aria-selected={true}
        className="tab"
        data-title="Monday"
        href="/?schedule=monday"
        id="monday-tab"
        onClick={[Function]}
        role="tab"
      >
        Monday
      </a>
      <a
        aria-controls="tuesday"
        aria-selected={false}
        className="tab"
        data-title="Tuesday"
        href="/?schedule=tuesday"
        id="tuesday-tab"
        onClick={[Function]}
        role="tab"
      >
        Tuesday
      </a>
      <a
        aria-controls="wednesday"
        aria-selected={false}
        className="tab"
        data-title="Wednesday"
        href="/?schedule=wednesday"
        id="wednesday-tab"
        onClick={[Function]}
        role="tab"
      >
        Wednesday
      </a>
      <a
        aria-controls="thursday"
        aria-selected={false}
        className="tab"
        data-title="Thursday"
        href="/?schedule=thursday"
        id="thursday-tab"
        onClick={[Function]}
        role="tab"
      >
        Thursday
      </a>
      <a
        aria-controls="friday"
        aria-selected={false}
        className="tab"
        data-title="Friday"
        href="/?schedule=friday"
        id="friday-tab"
        onClick={[Function]}
        role="tab"
      >
        Friday
      </a>
      <a
        aria-controls="saturday"
        aria-selected={false}
        className="tab"
        data-title="Saturday"
        href="/?schedule=saturday"
        id="saturday-tab"
        onClick={[Function]}
        role="tab"
      >
        Saturday
      </a>
      <a
        aria-controls="sunday"
        aria-selected={false}
        className="tab"
        data-title="Sunday"
        href="/?schedule=sunday"
        id="sunday-tab"
        onClick={[Function]}
        role="tab"
      >
        Sunday
      </a>
    </nav>
    <div
      aria-hidden={false}
      aria-labelledby="monday-tab"
      className="tabpanel"
      id="monday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="tuesday-tab"
      className="tabpanel"
      id="tuesday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="wednesday-tab"
      className="tabpanel"
      id="wednesday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="thursday-tab"
      className="tabpanel"
      id="thursday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="friday-tab"
      className="tabpanel"
      id="friday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="saturday-tab"
      className="tabpanel"
      id="saturday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="sunday-tab"
      className="tabpanel"
      id="sunday"
      role="tabpanel"
    >
      <table
        className="daily-schedule"
      >
        <thead>
          <tr
            className="timeslot"
          >
            <th
              className="timeslot_time"
            >
              Time Slot
            </th>
            <th
              className="timeslot_recurrence"
            >
              Recurrence
            </th>
            <th
              className="timeslot_collection"
            >
              Collection
            </th>
            <th
              className="timeslot_actions"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      className="tab-footer"
    >
      <button
        className="linkButton"
        onClick={[Function]}
      >
        Add new time slot
      </button>
    </div>
  </div>
</section>
`;
